### PR TITLE
Expandable Block

### PR DESCRIPTION
**Pull request by philipjohn**
Editor View | Frontend View
-------- | ----------
expandable-editor | expandable-frontend

Behavior

**Editor**

- Block is collapsed by default
- Block defaults to 300px height
- Selecting block reveals both "show more" and "show less" textboxes for editing
- Selecting block allows sidebar to be opened to control block height
- Selecting a child of the block expands the block content
- Block permits sizing by px, %, or vh units

**Frontend**

- Block is collapsed by default
- PHP is used to decorate saved content with aria attributes and a unique ID to link the button to its controlled container
- Clicking the button toggles visibility
- Focusing an element within the container forces it open, for accessibility

**Remaining Issues / Future Improvements**

- Alter color of carat based on text color. [Possible when loading an svg as a background image using CSS](https://codepen.io/sosuke/pen/Pjoqqp), but not (I believe) if color is set with custom variable.
- Permit converting a Group Block into an Expandable

### CODE REVIEW

File: inc/report.php [hunk 1]
```diff
@@ -35,3 +36,28 @@ function register_post_type() {
...
+ * @param bool|string[]            $allowed_block_types  Array of block type slugs, or boolean to enable/disable all.
+ * @param \WP_Block_Editor_Context $block_editor_context The current block editor context.
+ *
+ * @return bool|string[] Filtered allowed blocks list.
+ */
+function allowed_block_types( $allowed_block_types, $block_editor_context ) {
+
+	// Ensure all our blocks are allowed and not excluded by the theme.
+	$ar_blocks = [
+		'expandable'
```
Comment (kadamwhite) [inline inc/report.php:52]:
```suggestion
		'wikimedia-annual-report/expandable'
```
Needs to be the full block name

Comment (kadamwhite) [reply inc/report.php:52]: Nevermind that's addressed below

File: src/blocks/expandable/view.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// Add a click event listener to each expandable block.
+// When the button is clicked, toggle the is-expanded class on the content.
+document
+	.querySelectorAll( '.wp-block-wikimedia-annual-report-expandable' )
+	.forEach( ( block ) => {
+		const button = block.querySelector( '.expandable-button' );
+		const content = block.querySelector( '.expandable-content' );
+
+		button.addEventListener( 'click', () => {
```
Comment (goldenapples) [inline src/blocks/expandable/view.js:32]: I think we might need more javascript for interactivity than this to meet accessibility guidelines. At the minimum, from what I'm seeing, the "aria-expanded" attribute should be set on the parent element to give screen readers context.

https://www.w3.org/WAI/ARIA/apg/patterns/accordion/

Comment (kadamwhite) [reply src/blocks/expandable/view.js]: I read this as saying that the aria-expanded should go on the *button* — "....the header button has `aria-expanded` set to `true`", etc. I've therefore provisionally made changes here, but a final review might be good.

**Review by kadamwhite — APPROVED** [review]
@goldenapples @philipjohn Could use a final once-over, I adjusted the frontend logic a bit to better match my interpretation of the Aria resources Than shared and fixed a couple editor-side interaction bugs. I've filled out the description of the issue.

**Pull request merged**